### PR TITLE
Fix payments bug and add iot transfer support

### PR DESCRIPTION
**Pull request by Perronef5**

### CODE REVIEW

File: src/features/payment/PaymentScreen.tsx [hunk 1]
```diff
@@ -351,25 +348,17 @@ const PaymentScreen = () => {
...
-      }
-
-      if (ticker === 'IOT') {
-        const hasEnoughNetwork =
-          networkBalance.minus(paymentState.networkFee).integerBalance >= 0
-        const hasEnoughIOT =
-          iotSolBalance - paymentState.totalAmount.floatBalance.valueOf() >= 0
-        if (!hasEnoughNetwork) return [true, networkBalance.type.ticker]
-        if (!hasEnoughIOT) return [true, 'IOT']
+      if (l1Network === 'helium' || l1Network === 'solana') {
```
Comment (tyler-whitman) [inline src/features/payment/PaymentScreen.tsx:351]: When does `l1Network` not equal `helium` or `solana`?

File: src/features/payment/PaymentScreen.tsx [hunk 2]
```diff
@@ -351,25 +348,17 @@ const PaymentScreen = () => {
...
-        const hasEnoughNetwork =
-          networkBalance.minus(paymentState.networkFee).integerBalance >= 0
-        const hasEnoughMobile =
-          mobileSolBalance - paymentState.totalAmount.floatBalance.valueOf() >=
-          0
-        if (!hasEnoughNetwork) return [true, networkBalance.type.ticker]
-        if (!hasEnoughMobile) return [true, 'MOBILE']
-      }
-
-      if (ticker === 'IOT') {
```
Comment (tyler-whitman) [inline src/features/payment/PaymentScreen.tsx:366]: Dont need this check anymore?

**Review by tyler-whitman — APPROVED** [review]

File: src/features/payment/PaymentScreen.tsx (hunk 1 above)
Comment (Perronef5) [reply src/features/payment/PaymentScreen.tsx]: Oops should only be checking `helium` here

File: src/features/payment/PaymentScreen.tsx (hunk 2 above)
Comment (Perronef5) [reply src/features/payment/PaymentScreen.tsx:366]: No cause IOT only exists in `solana` world

**Pull request merged**